readtags-cmd/readtags-cmd.c

```diff
@@ -136,40 +136,59 @@ static void listTags (int pseudoTags)
 static const char *const Usage =
 	"Find tag file entries matching specified names.\n\n"
 	"Usage: \n"
-	"    %s -h\n"
-	"    %s [-ilp] [-n] "
+	"    %s -h | --help\n"
+	"        Print this help message.\n"
 #ifdef READTAGS_DSL
-	"[-Q EXP] "
+	"    %s -HQ| --help-expression filter\n"
+	"        Print available operators in filter expression.\n"
 #endif
-	"[-s[0|1]] [-t file] [-] [name(s)]\n\n"
+	"    %s [OPTIONS] ACTION\n"
+	"        Do the specified action.\n"
+	"Actions:\n"
+	"    -l | --list\n"
+	"        List regular tags.\n"
+	"    [-] name...\n"
+	"        List regular tags matching the name.\n"
+	"        \"-\" indicates arguments after this as NAME even if they start with -.\n"
+	"    -D | --list-pseudo-tags\n"
+	"        List pseudo tags\n"
 	"Options:\n"
-	"    -d           Turn on debugging output.\n"
-	"    -D           List all pseudo tags.\n"
-	"    -e           Include extension fields in output.\n"
-	"    -h           Print this help message.\n"
-	"    -i           Perform case-insensitive matching.\n"
-	"    -l           List all tags.\n"
-	"    -n           Allow print line numbers if -e option is given.\n"
-	"    -p           Perform partial matching.\n"
+	"    -d | --debug\n"
+	"        Turn on debugging output.\n"
+	"    -e | --extension-fields\n"
+	"        Include extension fields in output.\n"
+	"    -i | --icase-match\n"
+	"        Perform case-insensitive matching.\n"
+	"    -n | --line-number\n"
+	"        Allow print line numbers if -e option is given.\n"
+	"    -p | --prefix-match\n"
+	"        Perform prefix matching.\n"
+	"    -t TAGFILE | --tag-file TAGFILE\n"
+	"        Use specified tag file (default: \"tags\").\n"
+	"    -s[0|1|2] | --override-sort-detection METHOD\n"
+	"        Override sort detection of tag file.\n"
+	"        METHOD: unsorted|sorted|foldcase\n"
 #ifdef READTAGS_DSL
-	"    -Q EXP       Filter the result with EXP.\n"
+	"    -Q EXP | --filter EXP\n"
+	"        Filter the tags listed by ACTION with EXP before displaying.\n"
 #endif
-	"    -s[0|1|2]    Override sort detection of tag file.\n"
-	"    -t file      Use specified tag file (default: \"tags\").\n"
-	"    -            Treat arguments after this as NAME even if they start with -.\n"
+	"\n"
 	"Note that options are acted upon as encountered, so order is significant.\n";
```

Comment:
I'm not sure if we still need this, since we didn't say in the help message that there can be multiple actions, and it's already clear that the user should put options before the action. If the order of options doesn't make a difference, we can safely get rid of this line.

Reply:
I agree with you. I deleted the line.